src/homebridge-roku.js

```diff
@@ -17,28 +23,41 @@ class RokuAccessory {
       throw new Error(`An ip address is required for plugin ${this.name}`);
     }
 
-    this.appMap = config.appMap;
     this.info = config.info;
+    this.inputs = config.inputs;
     this.roku = new Client(config.ip);
     this.services = [];
 
     this.volumeIncrement = config.volumeIncrement || DEFAULT_VOLUME_INCREMENT;
     this.volumeDecrement = config.volumeDecrement || this.volumeIncrement;
 
     this.muted = false;
-    this.volumeLevel = 50;
-    this.poweredOn = false;
+    this.poweredOn = Characteristic.Active.INACTIVE;
+
+    this.buttons = {
+      [Characteristic.RemoteKey.REWIND]: keys.REVERSE,
+      [Characteristic.RemoteKey.FAST_FORWARD]: keys.FORWARD,
+      [Characteristic.RemoteKey.NEXT_TRACK]: keys.REVERSE,
+      [Characteristic.RemoteKey.PREVIOUS_TRACK]: keys.FORWARD,
+      [Characteristic.RemoteKey.ARROW_UP]: keys.UP,
+      [Characteristic.RemoteKey.ARROW_DOWN]: keys.DOWN,
+      [Characteristic.RemoteKey.ARROW_LEFT]: keys.LEFT,
+      [Characteristic.RemoteKey.ARROW_RIGHT]: keys.RIGHT,
+      [Characteristic.RemoteKey.SELECT]: keys.SELECT,
+      [Characteristic.RemoteKey.BACK]: keys.BACK,
+      [Characteristic.RemoteKey.EXIT]: keys.HOME,
+      [Characteristic.RemoteKey.PLAY_PAUSE]: keys.PLAY,
+      [Characteristic.RemoteKey.INFORMATION]: keys.INFO,
+    };
 
     this.setup();
   }
 
   setup() {
-    this.services.push(this.setupAccessoryInfo());
-    this.services.push(this.setupSwitch());
-    this.services.push(this.setupMute());
-    this.services.push(this.setupVolumeUp());
-    this.services.push(this.setupVolumeDown());
-    this.services.push(...this.setupChannels());
+    this.setupAccessoryInfo();
```

Comment:
I think these are easier to test when they are pure functions (returning their config and not modifying the services array). Can you refactor these setup functions so that only this setup method touches the services array?

Reply:
OK.